src/todo/UI.lua

```diff
@@ -145,13 +145,24 @@ function todo.create_add_edit_frame(player)
     })
 
 
-    local players, _ = todo.get_player_list()
-    table.add({
-        type = "drop-down",
-        name = "todo_add_assignee_drop_down",
-        items = players,
-        selected_index = 1
-    })
+    local players, _, c = todo.get_player_list()
+    -- The count is off by 1 so if the result is 2 we get only one player
+    if( todo.auto_assign(player) and c == 2 ) then
+        table.add({
+            type = "drop-down",
+            name = "todo_add_assignee_drop_down",
+            items = players,
+            selected_index = 2
```

Comment:
I would only do the index decision in the if and then use the resulting value in the add block. This makes it a little bit more clear where the difference is and what is actually decided.